### PR TITLE
Cluster check result modal

### PR DESCRIPTION
**Pull request by arbulu89**
Implementation of the cluster checks result modal. It provides the ansible runner checks result in the cluster page (only the latest execution, we don't have an historical view yet), showing all the checks results by host.

It consumes the ARA record `trento-results` from ARA, and in combination with metadata information, it displays the modal.

I have done some changes in the ARA records json structures to be more meaningful. (https://github.com/trento-project/trento/commit/08e3f5d6f580a3332c6ddf22b5355beda325e4ce)

Besides that, some new models for the checks are added, to handle this new Json structure.

**Disclaimer:** The results are changed once the runner is executed, so even though we change the selected checks from the settings modal, the update of the results is not immediate.

Here a short demo:
checks_result

### CODE REVIEW

File: web/services/checks.go [hunk 1]
```diff
@@ -62,3 +69,39 @@ func (c *checksService) GetChecksCatalogByGroup() (map[string]map[string]*models
...
+	if err != nil {
+		return nil, err
+	}
+
+	mapstructure.Decode(record.Value, &checkData)
+
+	return checkData.Groups, nil
+}
+
+func (c *checksService) GetChecksResultByCluster(clusterName string) (*models.Results, error) {
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply web/services/checks.go:95]: I will need to have a look, as ansible is not aware of this add, so the inventory file must be changed.
Let's see how I can do it.

[the remark this answers is not in the document]
Comment (arbulu89) [reply web/services/checks.go]: Fixed in: https://github.com/trento-project/trento/pull/189/commits/b18dbc8f871de8276945c543a12c4c75d2fef981

**Comment by arbulu89** [comment]
Issues and comments addressed. Besides this, I have implemented the notifications to the users for the scenario where some node is not reachable by ansible. It looks like this:
image

**Comment by diegoakechi** [comment]
@arbulu89  Some small UX refinements suggestions here. 

I think the list of checks is already big, so I suggest we have sorting and filtering options on the modal too, so the user can find the failed test easier than scrolling it completely. Maybe a summary on the header for each node could make sense too. 

Another point is that, on the first video, I might have spotted a bug. The main counters on the header of the main screen presents 0 success, while when checking the results we can see we have successes there.  Let me know if a new issue is needed for that.

**Comment by arbulu89** [comment]
> @arbulu89 Some small UX refinements suggestions here.
> 
> I think the list of checks is already big, so I suggest we have sorting and filtering options on the modal too, so the user can find the failed test easier than scrolling it completely. Maybe a summary on the header for each node could make sense too.
> 
> Another point is that, on the first video, I might have spotted a bug. The main counters on the header of the main screen presents 0 success, while when checking the results we can see we have successes there. Let me know if a new issue is needed for that.

Thank you for checking it out.

The filtering is something we have in mind, but It won't come in this PR. It requires some important frontend changes that are not simple to execute. We will work on that in future iterations.
About the main page and the modal, the synchronization of them is not implemented either, it will come in future PRs. In any case, the page summary window doesn't show the check results data, it shows the nodes state summary, which is the number of failed/working nodes in this cluster

**Review by fabriziosestito — APPROVED** [review]

**Comment by fabriziosestito** [comment]
Thanks!

**Pull request merged**